src/OpenTelemetry.Instrumentation.Http/MeterProviderBuilderExtensions.cs

```diff
@@ -64,10 +64,18 @@ public static class MeterProviderBuilderExtensions
             });
         }
 #else
-        builder.AddMeter(HttpHandlerMetricsDiagnosticListener.MeterName);
+        if (HttpHandlerMetricsDiagnosticListener.IsNet8OrGreater)
```

Comment:
I think I found code enabling `http.client.request.duration` instrumentation mentioned in #4932.

@antonfirsov mentioned the following:

> Note that HttpMetricsEnrichmentContext is a low-level primitive that needs a callback registration call for each HttpRequestMessage, and the assumption was that higher-level libraries will build their own (more convenient) enrichment experiences around it. 

I am trying to understand - are you planning to add a feature to open telemetry, supporting http client request enrichment? If this is not something planned to be part of open telemetry, then is expectation that enrichments will be attached using custom http client handler, with code passing enriching callback?

```csharp
HttpMetricsEnrichmentContext.AddCallback(message, (ctx) =>
    {
        ctx.AddCustomTag("MyCustomTag", "MyCustomValue");
    });
```

@vishweshbankwar

Reply:
> I think I found code enabling http.client.request.duration instrumentation mentioned in #4932.

I would assume this alone will not enable the instruments, and you would also need `AddInMemoryExporter`. @vishweshbankwar how does this work?

Reply:
> I am trying to understand - are you planning to add a feature to open telemetry, supporting http client request enrichment?

I think it would be the best if `opentelemetry-dotnet` exposed an `HttpMetricsEnrichmentContext`-based replacement of their current enrichment API-s:

https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.Http#enrich-httpclient-api

Designing a global callback-registration API is an opinionated area. It depends on how users manage their HttpClients. (ie. an API on HttpClient doesn't help much, if you use IHttpClientFactory.)

@JamesNK any thoughts?

Reply:
We also need to think about cases, when http client is used by referenced package, which does not expose it, but we still want to collect these metrics and attach custom tags. Not sure if this feature is available.

Reply:
What are you trying to achieve? What are the scenarios?

If someone has configured a callback to a `HttpRequestMessage` with `HttpMetricsEnrichmentContext.AddCallback`, then the tags added should automatically flow through into the `http.client.request.duration` counter. There is no need to add features to OpenTelemetry.

`HttpMetricsEnrichmentContext.AddCallback` is a low-level API but dotnet/extensions is adding support for configuring enrichers via DI.